test/linalg/diagonal.jl

```diff
@@ -2,12 +2,45 @@
 
 using Base.Test
 import Base.LinAlg: BlasFloat, BlasComplex, SingularException
+import Base: size, linearindexing, LinearSlow, getindex, length, setindex!, ndims
 
 debug = false
 
 n=12 #Size of matrix problem to test
 srand(1)
 
+#we'll use this to test A_ldiv_B!
+type SparseVec{T} <: AbstractVector{T}
```

Comment:
I couldn't get it to work and @vtjnash suggested this. I can redo the tests if necessary.